### DISCUSSION TITLE
Best way to get contents of a file with RTR

### DISCUSSION THREAD

**Discussion by Emarples** [CrowdStrike/falconpy#529] [Q&A] [answered]
Hi! 
I'm trying to transition my team from using the GUI to RTR and download windows event logs, to doing through the API to speed up the process. We have a script that writes the logs onto a file on the host, but I'm having trouble getting them from there. I've tried the get command and even though it succeeded I'm not sure how to actually download the file without the GUI. That leaves me with the following questions. 

1) how does using the get command work with the API and is there anyway to download the file after running it (without using the CS GUI)? 

2) If that's not possible, do you have any suggestions for getting the contents of a file on a host through the RTR API? 

Any help is greatly appreciated, thanks!

**Comment by jshcodes** [comment, marked as answer]
Hi @Emarples!

1) Get retrieves the file off of the host and stores it within the CrowdStrike cloud for retrieval. An example of how to use this functionality can be found in the "PID dump" sample located [here](https://github.com/CrowdStrike/falconpy/blob/main/samples/rtr/pid-dump/rtr_dump_memory.py#L300). 

2) You should be able to download the file, but if you are just interested in retrieving the contents of the file, you could try something similar to what is performed in the [My Little RTR example](https://github.com/CrowdStrike/falconpy/blob/main/samples/rtr/pony/my_little_rtr.py#L214).

Check these out and let us know if there are any questions! 😃

**Reply by Emarples** [reply]
Awesome, thank you!!
